nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -126,28 +128,70 @@ BtNavigator::on_configure(const rclcpp_lifecycle::State & /*state*/)
   blackboard_->set<int>("number_recoveries", 0);  // NOLINT
 
   // Get the BT filename to use from the node parameter
-  std::string bt_xml_filename;
-  get_parameter("bt_xml_filename", bt_xml_filename);
+  get_parameter("bt_xml_filenames", bt_xml_filenames_);
+  get_parameter("bt_xml_ids", bt_xml_ids_);
+
+  if (bt_xml_filenames_.empty()) {
+    RCLCPP_ERROR(get_logger(), "Couldn't find any XML file");
+    return nav2_util::CallbackReturn::FAILURE;
+  }
+
+  if (!loadBehaviorTree(bt_xml_ids_[0])) {
+    RCLCPP_ERROR(get_logger(), "Error loading XML file: %s", bt_xml_filenames_[0].c_str());
+    return nav2_util::CallbackReturn::FAILURE;
+  }
+
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+bool
+BtNavigator::loadBehaviorTree(const std::string & bt_id)
+{
+  // Use previous BT if it is the existing one
+  if (current_bt_id_ == bt_id) {
+    return true;
+  }
+
+  // Get the BT XML file corresponding to id
+  auto bt_id_it = std::find(bt_xml_ids_.begin(), bt_xml_ids_.end(), bt_id);
```

Comment:
You should store this as an `unordered_map` for easier semantics. About half of this logic goes away.

Reply:
Removed because it is not used now